File: 3rd Sem/DSA/TowerOfHanoi.java
import java.util.Scanner;

public class TowerOfHanoi{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("enter the  number of disk");
        int n = sc.nextInt();
        TOH(n, 'S', 'D', 'T');

    }

    static void TOH(int n, char A, char B, char C) {

        if (n == 1) {
            System.out.println("From " + A + " to " + B);
        } else {
            TOH(n - 1, A, C, B);
            TOH(1, A, B, C);
            TOH(n - 1, C, B, A);
        }
    }
}

File: 5th sem/AI/lab 6/towerofhanoi.py
def tower_of_hanoi(n, from_rod, to_rod, aux_rod):
    """
    Args:
        n (int): The number of disks.
        from_rod (str): The rod where disks are initially located.
        to_rod (str): The rod where disks need to be moved.
        aux_rod (str): The auxiliary rod used for temporary storage.
    """
    if n == 1:
        print(f"Move disk 1 from {from_rod} to {to_rod}")
        return
    tower_of_hanoi(n-1, from_rod, aux_rod, to_rod)
    print(f"Move disk {n} from {from_rod} to {to_rod}")
    tower_of_hanoi(n-1, aux_rod, to_rod, from_rod)

def main():
    # Get user input
    n = int(input("Enter the number of disks: "))
    
    # Solve the Tower of Hanoi problem
    tower_of_hanoi(n, 'A', 'C', 'B')

if __name__ == "__main__":
    main()